lesson3.md

```diff
@@ -0,0 +1,841 @@
+<!-- .element: class="fragment" -->
+
+
+---
+
+
+### Quick Quiz (Cont.)
+
+
+I can only pay 100€, is it enough?
+
+
+The cinema will only allow us to enter if we are less than 30 people and if we pay more than 100€, will we get in?
+<!-- .element: class="fragment" -->
+
+
+---
+
+
+### Recap
+
+| Group | Operators | Example |
+| ----- | --------- | ------- |
+| Numerical Operators | `+ - * /` | `5 + 4 * 3`<br/>`7 / 2 - 2`<br/>`"Hello" + " World"` |
+| Comparison Operators | `=== !== < > <= >=` | `temperature !== 25`<br/>`age >= 18` |
+| Logical Operators | &#124;&#124; `&& !` | `a && !b`<br />`x >= 5 && x < 15` |
+| **Combined Operators** | `+= -= *= /= ++ --` | `a *= 2`<br />`count++` |
+
+
+---
+
+
+### Excursus: Combined Operators
+
+This
+
+```js
+let myAge = 42;
+myAge = myAge + 5;
+```
+
+Is the same as this
+
+```js
+let myAge = 42;
+myAge += 5;
+```
+
+
+---
+
+
+### Quiz: What's the price?
+
+```js
+let price = 10;
+
+// there's a 50 percent sale!
+price /= 2;
+
+// price is ??
+```
+
+Result: `5`
+<!-- .element: class="fragment" -->
+
+
+---
+
+
+### Operator ++ and --
+
+Operator `++` and `--` increase (or decrease) the value of a variable by 1
+
+```js
+let myAge = 42;
+myAge++; // same as myAge = myAge + 1
+```
+
+Combined operators only work with variables!
+
+---
+
+
+<!-- .slide: id="lesson3:if" -->
+# Conditional Statements
+
+
+---
+
+
+### Conditional Statements
+
+Up until now, we wrote code line by line.
+
+But what if I want to run some code only if the user is logged in?
+
+
+---
+
+
+### If statement
+
+```js
+if (condition) {
+  // block of code that
+  // will run ONLY if
+  // condition is true
+}
+```
+
+
+---
+
+
+### If statements
+
+```js
+if (condition) {
+  // your code here
+}
+```
+
+* `condition` must be a boolean
+* The code in the block is __only__ executed if the boolean is `true`
+* If the boolean is false, the code block is _not_ executed
+
+
+---
+
+
+### if Statements - Examples
+
+```js
+if (food === 'pizza') {
+  console.log('My favorite!');
+}
+```
+
+```js
+if (money < 1000000) {
+  console.log('You are not rich :p');
+}
+```
+
+```js
+if (canSpeakRussian) { // same as: canSpeakRussian === true
+  console.log('Привет!');
+}
+```
+
+
+---
+
+
+### Code Blocks: {}
+
+Code blocks contain the code between `{` and `}`, and should be indented for better readability:
+
+```js
+if (language === "German") {
+  // In VSCode, you can indent code by pressing the "Tab" key,
+  // or right-click and choose "Format Document"
+  console.log("Guten Tag!");
+}
+```
+
+
+---
+
+
+### Code Blocks: {}
+
+All variables created in code blocks will disappear after closing the block (technical term: “block scope”)
+
+```js
+let name = "John Doe";
+
+if (name === "John Doe") {
+  let greeting = "Hi dude";
+}
+if (name === "Mary Doyle") {
+  let greeting = "Hello my dear";
+}
+
+console.log(greeting); // ERROR!!!!
+```
+
+
+---
+
+
+### Code Blocks: {}
+
+Correct solution:
+
+```js
+let name = "John Doe";
+
+let greeting;
```

Comment:
I, personally, would not leave this variable uninitialized.

Reply:
Me too, but if they don't cover all branches they will learn that the value of an uninitialized variable is `undefined`. win/win situation.